### ISSUE TITLE
Demonstrate generated source files

### ISSUE THREAD

**Issue by lacasseio** [nokeedev/gradle-native#623]
To complement globing and cherry-picking, we can generate sources:

```
def generatorTask = tasks.register('generateSources', Copy) {
    from(zipTree('sources.zip')) {
        into('sources') { include '**/*.c' }
        into('includes') { include '**/*.h' }
    }
    destinationDir = layout.buildDirectory.dir('generated/main').get().asFile
}

application {
    cSources.setFrom(generatorTask.map { "${destinationDir}/sources" })
    privateHeaders.setFrom(generatorTask.map { "${destinationDir}/includes" })
}
```

It's important to note there are several ways to implement a generator. This particular generator is one way to extract files from a zip file. However, you could use `zipTree` directly as inputs to `cSources` and `privateHeaders`. This is a demonstration. Usually the `Copy` task would be replaced by a custom task implementation.

**Comment by Ladalz** [comment]
Is generator a common knowledge if not we also need to add to the FYI "what is a generator for sources and when to use them"

**Comment by lacasseio** [comment]
Good point, the generator is the abstract process to generate the sources. I think we can have a quick mention when we gloss over the generator task implementation. Most users generally refer to _generated sources_ which stand by themselves for _some process_ leading to _additional sources_. It could be a good idea to include it in [our terminology chapter](https://docs.nokee.dev/manual/terminology.html) which is inspired by [Gradle terminology for dependency management](https://docs.gradle.org/current/userguide/dependency_management_terminology.html). I'm unsure if we should have a focused terminology chapter but for now, we can bunch them all in the same chapter and split them later when it makes sense.

**Comment by lacasseio** [comment]
In other build systems, they use similar terminology. For example, in CMake, they have a section in the tutorial about [generated files](https://cmake.org/cmake/help/latest/guide/tutorial/Adding%20a%20Custom%20Command%20and%20Generated%20File.html) and Meson has a section about [generating sources](https://mesonbuild.com/Generating-sources.html). When reading their documentation, we have to be careful because the term _generator_ has a different meaning in some contexts.

In general, _generated sources_ is the result the user wants and _generating sources_ (or _generator_) is the process to arrive at the result. There are multiple ways to get the same result which is why we should focus more on the result and less on the process. The result is more in the control of Nokee while the process is more in the control of Gradle.
